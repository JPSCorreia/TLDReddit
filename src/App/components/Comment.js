import React from 'react'
import ReactMarkdown from 'react-markdown'
import * as App from '../App'

function Comment(props) {

  console.log(props.commentData)
  return (
    <div className="comment">
      <div className='author-info'>
        <div className='author'>{props.commentData.author}</div>
        <span className='separator'> &nbsp; </span>
        <div className='points'>{props.commentData.ups} points</div>
        <span className='separator'> &nbsp; </span>
        <div className='comment-created-when'>{App.convertUnixToDate(props.commentData.created_utc)}</div>
      </div>
      <div className='comment-body'><ReactMarkdown>{props.commentData.body}</ReactMarkdown></div>
    </div>
  );
}

export default Comment;