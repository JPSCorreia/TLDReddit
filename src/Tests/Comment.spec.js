import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import React from 'react';
import Comment from '../App/components/Comment';
configure({adapter: new Adapter()});


// Function to create test properties, spread props in the end allows changing of attributes.
const createTestProps = (props) => {
return {
  commentData: {
    created_utc: 1634130100,
    author: 'test_author',
    ups: 5000,
    body: "this is a test for comment's body",
  },
  topicId: 'r/worldnews-1',
  dataKey: 2,
  ...props
}
}


describe('Comment Component Testing:', () => {


  it('Should render with FAKE test data', () => {

    const props = createTestProps();
    const component = shallow(<Comment {...props} />);
    expect(component).toMatchSnapshot();

  });


  it('Should render even with UNDEFINED test data', () => {

    const props = createTestProps({
      commentData: {
        created_utc: undefined,
        author: undefined,
        ups: undefined,
        body: undefined
      },
      topicId: undefined,
      dataKey: undefined
    })

    const component = shallow(<Comment {...props} />);
    expect(component).toMatchSnapshot();

  });


  it("renders Comment component header with id attribute being passed properly", () => {
    
    const props = createTestProps();
    const header = (<div class="comment" id='comment-r/worldnews-1-2'></div>);
    const component = shallow(<Comment {...props} />);
    expect(component.contains(header)).toEqual(true);

  });

});

