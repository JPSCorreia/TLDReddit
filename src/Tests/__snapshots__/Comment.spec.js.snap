// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Component Testing: Should render even with UNDEFINED test data 1`] = `
<div
  className="comment"
  id="comment-undefined-undefined"
>
  <div
    className="author-info"
  >
    <div
      className="author"
    />
    <span
      className="separator"
    >
         
    </span>
    <div
      className="points"
    >
       points
    </div>
    <span
      className="separator"
    >
         
    </span>
    <div
      className="comment-created-when"
    >
      <n
        ago={false}
        calendar={false}
        decimal={false}
        element={null}
        filter={[Function]}
        fromNow={true}
        interval={60000}
        local={false}
        onChange={[Function]}
        titleFormat=""
        toNow={false}
        trim={false}
        unit={null}
        unix={true}
        utc={false}
        withTitle={false}
      />
    </div>
  </div>
  <div
    className="comment-body"
  />
</div>
`;

exports[`Comment Component Testing: Should render with FAKE test data 1`] = `
<div
  className="comment"
  id="comment-r/worldnews-1-2"
>
  <div
    className="author-info"
  >
    <div
      className="author"
    >
      test_author
    </div>
    <span
      className="separator"
    >
         
    </span>
    <div
      className="points"
    >
      5000
       points
    </div>
    <span
      className="separator"
    >
         
    </span>
    <div
      className="comment-created-when"
    >
      <n
        ago={false}
        calendar={false}
        decimal={false}
        element={null}
        filter={[Function]}
        fromNow={true}
        interval={60000}
        local={false}
        onChange={[Function]}
        titleFormat=""
        toNow={false}
        trim={false}
        unit={null}
        unix={true}
        utc={false}
        withTitle={false}
      >
        1634130100
      </n>
    </div>
  </div>
  <div
    className="comment-body"
  >
    this is a test for comment's body
  </div>
</div>
`;
